bentoml/_internal/runner/runnable.py

```diff
@@ -18,10 +19,9 @@
 logger = logging.getLogger(__name__)
 
 RUNNABLE_METHOD_MARK: str = "_bentoml_runnable_method"
-# BatchDimType: t.TypeAlias = t.Tuple[t.List[int] | int, t.List[int] | int] | int
-BatchDimType: t.TypeAlias = t.Union[
-    t.Tuple[t.Union[t.List[int], int], t.Union[t.List[int], int]], int
-]
+# not usable because attrs is unable to resolve the type at runtime
+# BatchDimType: t.TypeAlias = tuple[list[int] | int, list[int] | int] | int
+BatchDimType: t.TypeAlias = t.Union[t.Tuple[t.Union[t.List[int], int], int], int]
```

Comment:
Surprisingly this seems to work. We can't use the new type annotations because we're using cattrs structure / unstructure and it needs to resolve types at runtime.

Reply:
> Surprisingly this seems to work
It works because from Python 3.7, the annotation part won't be evaluated in runtime.

Reply:
I meant the `t.TypeAlias` bit

Reply:
Ah, nevermind, I understand.